Libs/MRML/Core/vtkMRMLTableSQLiteStorageNode.cxx

```diff
@@ -0,0 +1,269 @@
+  createTableQuery += "(";
+
+  std::string insertPreamble = "INSERT into ";
+  insertPreamble += this->TableName;
+  insertPreamble += "(";
+
+  //get the columns from the vtkTable to finish the query
+  vtkIdType numColumns = table->GetNumberOfColumns();
+  for(vtkIdType i = 0; i < numColumns; i++)
+    {
+    //get this column's name
+    std::string columnName = table->GetColumn(i)->GetName();
+    createTableQuery += columnName;
+    insertPreamble += "'" + columnName + "'";
+
+    //figure out what type of data is stored in this column
+    std::string columnType = table->GetColumn(i)->GetClassName();
+
+    if( (columnType.find("String") != std::string::npos) ||
+        (columnType.find("Data") != std::string::npos) ||
+        (columnType.find("Variant") != std::string::npos) )
+      {
+      createTableQuery += " TEXT";
+      }
+    else if( (columnType.find("Double") != std::string::npos) ||
+             (columnType.find("Float") != std::string::npos) )
+      {
+      createTableQuery += " REAL";
+      }
+    else
+      {
+      createTableQuery += " INTEGER";
+      }
+    if(i == numColumns - 1)
+      {
+      createTableQuery += ");";
+      insertPreamble += ") VALUES (";
+      }
+    else
+      {
+      createTableQuery += ", ";
+      insertPreamble += ", ";
+      }
+    }
+
+  //perform the create table query
+  vtkSQLiteQuery *query =
+    static_cast<vtkSQLiteQuery*>(database->GetQueryInstance());
+
+  query->SetQuery(createTableQuery.c_str());
+  cout << "creating the table" << endl;
+  if(!query->Execute())
+    {
+    vtkErrorMacro(<<"Error performing 'create table' query");
+    }
+
+  //iterate over the rows of the vtkTable to complete the insert query
+  vtkIdType numRows = table->GetNumberOfRows();
+  for(vtkIdType i = 0; i < numRows; i++)
+    {
+    std::string insertQuery = insertPreamble;
+    for (vtkIdType j = 0; j < numColumns; j++)
+      {
+      insertQuery += "'" + table->GetValue(i, j).ToString() + "'";
+      if(j < numColumns - 1)
+        {
+        insertQuery += ", ";
+        }
+      }
+    insertQuery += ");";
+    //perform the insert query for this row
+    query->SetQuery(insertQuery.c_str());
+    if(!query->Execute())
+      {
+      vtkErrorMacro(<<"Error performing 'insert' query");
+      }
+    }
+
+  //cleanup and return
+  query->Delete();
+  database->Close();
+  database->Delete();
+
+  vtkDebugMacro("WriteData: successfully wrote table to database: " << fullName);
+  return 1;
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLTableSQLiteStorageNode::InitializeSupportedReadFileTypes()
+{
+  this->SupportedReadFileTypes->InsertNextValue("SQLight database (.db)");
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLTableSQLiteStorageNode::InitializeSupportedWriteFileTypes()
+{
+  this->SupportedWriteFileTypes->InsertNextValue("SQLight database (.db)");
```

Comment:
Due to some reason, if you create a new table, saving as .db is not offered as an option.
Probably because only one storage node provides all the file extensions, and it's the vtkMRMLTableStorageNode.